### ISSUE TITLE
about the facial landmarks

### ISSUE THREAD

**Issue by lhr-30** [face-analysis/emonet#13]
Why does the training set data contain facial landmarks? It seems that the facial landmarks are not used during the training process, just helps to crop the image?

**Comment by lhr-30** [comment]
> Why does the training set data contain facial landmarks? It seems that the facial landmarks are not used during the training process, just helps to crop the image?

and another question is how do you get the facial landmarks cause i found facial landmarks data in the training set?

**Comment by HEzelda** [comment]
same here!

**Comment by lhr-30** [comment]
> Why does the training set data contain facial landmarks? It seems that the facial landmarks are not used during the training process, just helps to crop the image?

and about landmarks, the paper does not menssion the loss function about the landmarks, is there any loss function about the landmarks learning?

**Comment by ravikiranrao** [comment]
@lhr-30 @HEzelda  As far as I understand, this paper does not train landmarks, rather obtain the landmarks from the pre-trained FAN network. The landmarks obtained are thus treated as a heatmap for the attention map.

**Comment by lhr-30** [comment]
> @lhr-30 @HEzelda As far as I understand, this paper does not train landmarks, rather obtain the landmarks from the pre-trained FAN network. The landmarks obtained are thus treated as a heatmap for the attention map.

But how to get the pre-trained FAN network? Do you mean we get the hyperparameter of the pre-trained FAN network in another way but not in this trainning process?

**Comment by ravikiranrao** [comment]
> But how to get the pre-trained FAN network? Do you mean we get the hyperparameter of the pre-trained FAN network in another way but not in this trainning process?

@lhr-30 Technically, yes. This paper effectively trains only the green part in [Fig. 3](https://www.nature.com/articles/s42256-020-00280-0.epdf?sharing_token=8UEdDpr-K7UpoDi5SYjGLtRgN0jAjWel9jnR3ZoTv0NmJm5LtKLATZX-wg4Cg96PU_Xagw0dC67imfvbYZzOaFRYUs-6qIlabUXidsXjQRkYDNWUaArHiGy8zxyz2DozptfHYg3G_HumDAHYUwgV2PRTjKDd-8LKGzbMstnxIxY%3D). The FAN network in which authors refer in the paper is from [this](https://openaccess.thecvf.com/content_ICCV_2017/papers/Bulat_How_Far_Are_ICCV_2017_paper.pdf).

**Comment by lhr-30** [comment]
Now I have difficulty in reproducing this paper. Thr problem is that how to train the FAN part. In the emonet, we only get 'heatmap' but if we want to add the loss function we need to get the face landmarks. So how to transfrom the heatmap to face landmarks or do you have some other way to compute this part of loss function? Thank you.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "face-analysis/emonet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月29日(星期三) 下午5:13
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [face-analysis/emonet] about the facial landmarks (Issue #13)





  
But how to get the pre-trained FAN network? Do you mean we get the hyperparameter of the pre-trained FAN network in another way but not in this trainning process?
  
@lhr-30 Technically, yes. This paper effectively trains only the green part in Fig. 3. The FAN network in which authors refer in the paper is from this.

**Comment by ravikiranrao** [comment]
Sorry @lhr-30, I do not know much about optimizing the FAN yet. As a trivial suggestion, you can contact the authors directly.

**Comment by sssmost** [comment]
Any advance from this: how obtain facial landmarks from heatmap? The paper doesn`t report details about that.
